apps/backend/src/lib/rds.ts

```diff
@@ -337,4 +398,75 @@ export class RDS {
         // Sort schedules by index
         return Object.values(schedulesMapping).sort((a, b) => a.index - b.index);
     }
+
+    /**
+     * Retrieves the current session from the database using the provided refresh token.
+     *
+     * @param db - The database or transaction object to use for the query.
+     * @param refreshToken - The refresh token to search for in the sessions table.
+     * @returns A promise that resolves to the current session object if found, or null if not found.
+     */
+    static async getCurrentSession(db: DatabaseOrTransaction, refreshToken: string) {
+        return db.transaction((tx) =>
+            tx
+                .select()
+                .from(sessions)
+                .where(eq(sessions.refreshToken, refreshToken))
+                .then((res) => res[0] ?? null)
+        );
+    }
+
+    /**
+     * Creates a new session for a user in the database.
+     *
+     * @param db - The database or transaction object to use for the operation.
+     * @param userID - The ID of the user for whom the session is being created.
+     * @returns A promise that resolves to the created session object or null if the creation failed.
+     */
+    static async createSession(db: DatabaseOrTransaction, userID: string): Promise<Session | null> {
+        return db.transaction((tx) =>
+            tx
+                .insert(sessions)
+                .values({
+                    id: crypto.randomUUID(),
```

Comment:
You can specify a default value generator in the table definition instead of generating it here.